src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ImmutableDictionaryOfStringTValueConverter.cs

```diff
@@ -31,6 +31,11 @@ protected override void ConvertCollection(ref ReadStack state, JsonSerializerOpt
             state.Current.ReturnValue = GetCreatorDelegate(options)((Dictionary<string, TValue>)state.Current.ReturnValue!);
         }
 
+        internal override object Convert(object value, JsonSerializerOptions options)
```

Comment:
```suggestion
        internal override object ConvertToDictionary(object value, JsonSerializerOptions options)
```

This needs to be more descriptive.